### DISCUSSION TITLE
Mass-to-light Ratio in fitting with AGN models

### DISCUSSION THREAD

**Discussion by Bovornpratch** [mboquien/cigale#71] [Q&A]
Hi, 

I am performing SED fitting with CIGALE for Type-1 AGN and Type-2 AGN samples.  The fitting is done separately for each type of AGN by setting the AGN model (Skirtor) to different angles and setting the host galaxy model parameters to the same setting.

Out of curiosity (and through a question by our reviewer), I decided to examine the M/L ratio of the best-fitted host galaxy models.  To do so, took the best fitted models, remove the AGN and attenuation components (so that we are left with pure stellar) and integrate the H-band luminosity. Then I took the bayes.stellar.mstar to compute the M/L of the bestfit type-1 and type-2 AGN. 

I expect the best-fitted type-1 and type-2 AGN models to have similar M/L as a function of stellar age. However, what I found is that the M/L of type-1 AGN is lower by roughly 0.25 dex. Attached to this message is a plot showing the distribution of type-1 and type-2 AGN. Top left shows the log M/L, top right shows stellar age distributions. The vertical lines show the median values. Bottom left, shows the M/L and stellar age distribution, the scatter plot shows the median M/L in bins of stellar age.  bottom right panel shows the M/L ratio as a function of H-band luminosity.

A point here is that the average stellar mass of type-1 and type-2 AGN in our results also show a difference in the bayes stellar mass which is consistent with the difference in M/L. This suggest that the difference we see is due the difference in M/L of the stellar models used. In my settings, metallicity was set fixed. The IMF used is the same too.

I am hoping to open a discussion to what could be the cause here, and if what I am seeing is actually true.  My understanding is that the M/L ratio of a galaxy should not be different outside the same IMF and stellar age. It also should not depend on the redshift or normalization of the fitting.  Also, since I am looking at H-band, the M/L should not depend strongly on color (/age) either.  I apologize in advance if this is trivial or due to my misunderstanding of how SED fitting or CIGALE works.

cheers,

Klod (Bovornpratch)

![image](https://github.com/mboquien/cigale/assets/20829556/b11b40ac-f1bb-4c73-ae54-fe545bcf0fbc)

**Comment by mboquien** [comment]
I would be cautious mixing the Bayesian estimates of the stellar mass and estimates using the best-fit. What I would recommend is to modify the code to add a `compute_fnu()` call before the dust is added (look at the dust attenuation module, there is a couple of calls, which can serve as an inspiration). With this it is easy to add the M/L estimate with `add_info()`, which can then be estimated as any other parameter. That way the M/L would be computed consistently and there would also be uncertainties on it.

Another point is that there may be differences in the star formation history, which would impact the M/L. This may not necessarily be physical. For instance type-1 AGN can be problematic due to their UV emission, which is not always easy to separate from that of young stellar populations. The stellar age is an imperfect tracer of variations of the SFH.

**Comment by Guang91** [comment]
I agree with @mboquien. It can be misleading to mix best-fit and Bayes. 

It's not impossible that type 1 and type 2 have different M/L. We have a paper that found type 1 has lower mass than type 2, but we didn't study the M/L there. 
https://ui.adsabs.harvard.edu/abs/2019ApJ...878...11Z/abstract

**Comment by Bovornpratch** [comment]
Hi guys, 

Thanks a lot for the comments! I quickly made a plot switching to using the "best" instead and those trends weaken and the M/L is more consistent with each other overall.  I will also try @mboquien suggestion and see what I get.

@Guang91 Yes, I have seen the paper (and I am already citing it ;) ). Many studies seem to find that trend too (T1<T2). My concern was that if the M/L applied to type-1 and type-2 are different then the mass between the two will also be systematically different given the same amount of light.  My initial thought is that this shouldn't be the case anyway but I still felt like checking just to be sure following some comments I received. 

Thanks a lot for your help!
